server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalExtendedStats.java

```diff
@@ -41,6 +44,8 @@ public static Metrics resolve(String name) {
         }
     }
 
+    public static Set<String> metricNames = Stream.of(Metrics.values()).map(Metrics::name).collect(Collectors.toSet());
```

Comment:
This should probably not be mutable, and definitely not public mutable.  Same for other aggs.

Reply:
or, wait, does that `toSet()` stream collector return an immutable set?  I don't remember.  I'd still be more comfortable if it was private at least.